mlflow/xgboost/__init__.py

```diff
@@ -718,27 +731,73 @@ def infer_model_signature(input_example):
                 model_format=model_format,
             )
 
+        # Whether to automatically log the training dataset as a dataset artifact.
+        if _log_datasets:
+            try:
+                # create a CodeDatasetSource
+                context_tags = context_registry.resolve_tags()
+                source = CodeDatasetSource(
+                    mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                    mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                )
+
+                # create a dataset
+                import xgboost as xgb
+
+                # dmatrix has a get_data method added in 1.7. skip for earlier versions.
+                if Version(xgb.__version__) >= Version("1.7.0"):
```

Comment:
Are there other methods by which we can fetch the dataset is pandas / numpy? We've been logging input examples for XGBoost DMatrix datasets for ~ 2 years now, so I think there must be some solution - hopefully something similar?

Reply:
per offline discussion, we will only support dataset autologging for >1.7. We would need to do some patching+deepcopy hacking for earlier versions.